### PR TITLE
Add server_properties.yml

### LINKED ISSUE #13: Add server properties file
I think it will be correct if we add a file in which it will be possible to change the properties of the server, for example, its port, the maximum number of players, etc.

### PR DESCRIPTION
**Pull request by hydrostaticcog**
Resolves #13 

This PR aims to add a `server_properties.yaml` file to control port, host, server MOTD, maximum players allowed, and toggling the Blackfire Probe.

The default configuration is as follows:

```
port: 25565
ip: 127.0.0.1
motd: A Minecraft Server
use_blackfire: false
max_players: 20
```

I found that the `set_max_players` function was not added to set the maximum number of players, so I added it given the importance of it to the config file.

I also fixed some typos (capitalized `.blocks.Materials`) in WorldIO.py and a logging inconsistency with debug mode, and I added #46 reincarnated (uses releases.json to set `minecraft:brand` data).

### CODE REVIEW

File: classes/player/Player.py [hunk 1]
```diff
@@ -84,7 +91,7 @@ def send_join_packets(self, player: Player):
         # 'brand'
         NetworkController.send_packet_player(player.entity_id, PacketType.PLUGIN_MESSAGE, {
             'channel': 'minecraft:brand',
-            'data': 'McPy/0.0.1-alpha',
+            'data': 'McPy/' + version,
```
Comment (Geolykt) [inline classes/player/Player.py]: We could just as well calculate the brand data beforehand, that is before the method is run for the extra speed.

File: classes/utils/__init__.py [hunk 2]
```diff
@@ -4,3 +4,4 @@
 from . import Thread
 from . import Utils
 from . import Vector
+from . import Config
```
Comment (Geolykt) [inline classes/utils/__init__.py]: I dislike the red circle for absolutely no reason at all

File: classes/utils/Config.py [hunk 3]
```diff
@@ -18,18 +18,10 @@ class ConfigParser:
     :raises Exceptions.UnsupportedConfigType: if _type is not between 1 and 2
     """
 
-    def __init__(self, config_location, _type: int = 1):
+    def __init__(self, _type: int = 1):
+        config_location = 'server.properties'
```
Comment (Geolykt) [inline classes/utils/Config.py]: If anything we should end it with .yaml since that is what we are parsing it with, .properties might be a bit misleading

File: server.properties [hunk 4]
```diff
@@ -0,0 +1,5 @@
+port: 25565
+ip: 127.0.0.1
+motd: A Minecraft Server
+use_blackfire: false
+max_players: 20
```
Comment (Geolykt) [inline server.properties]: same as for classes/utils/__init__.py

File: server.properties [hunk 5]
```diff
@@ -0,0 +1,5 @@
+port: 25565
+ip: 127.0.0.1
```
Comment (Geolykt) [inline server.properties]: I would use 0.0.0.8 or whatever it was again

File: classes/mcPy/McPy.py [hunk 6]
```diff
@@ -20,17 +23,15 @@ def get_available_core():
...
-    try:
-        # noinspection PyUnresolvedReferences
-        from blackfire import probe  # Profiler: https://blackfire.io free with the Git Student Package
-    except ImportError:
-        BLACKFIRE_ENABLED = False
-        logging.info("Blackfire not installed: passing")
-    else:
+    if (config['use_blackfire'] == True):
         BLACKFIRE_ENABLED = True
+        from blackfire import probe
```
Comment (Geolykt) [inline classes/mcPy/McPy.py]: I would retain the try-except clause just for safety (never underestimate the end user)

File: classes/Server.py [hunk 7]
```diff
@@ -9,6 +9,9 @@
 from .network.PacketType import PacketType
 from .player.Player import PlayerManager
 from .utils.Scheduler import SchedulerManager
+from classes.utils.Config import ConfigParser
+
+config = ConfigParser.load_config(1)
```
Comment (Geolykt) [inline classes/Server.py]: Honestly, this is the wrong place to load it

**Review by Geolykt — COMMENTED** [review]
I like the overall Idea though

File: classes/Server.py (hunk 7 above)
Comment (hydrostaticcog) [reply classes/Server.py]: where should we load it instead?

Comment (Geolykt) [reply classes/Server.py]: Probably within the McPy.py file as that is where I would expect it to be loaded

File: classes/player/Player.py (hunk 1 above)
Comment (hydrostaticcog) [reply classes/player/Player.py]: So should we leave this the way it was for speed? And then just change it on a new update?

File: classes/Server.py (hunk 7 above)
Comment (hydrostaticcog) [reply classes/Server.py]: the `NetworkController.start_process()` called in the `start` function here requires the host and port arguments, so that is why I load the config here.

File: classes/utils/__init__.py (hunk 2 above)
Comment (hydrostaticcog) [reply classes/utils/__init__.py]: I don't understand what you mean

Comment (Geolykt) [reply classes/utils/__init__.py]: No newline at the end of file

File: classes/player/Player.py (hunk 1 above)
Comment (Geolykt) [reply classes/player/Player.py]: the version tag is already calculated beforehand, so we can bundle these two actions.

Honestly, this is some super wired nitpicking there

Comment (hydrostaticcog) [reply classes/player/Player.py]: It would probably just be easier if we followed what everyone else does and just make it `McPy`.

File: classes/Server.py (hunk 7 above)
Comment (hydrostaticcog) [reply classes/Server.py]: I'im going to try add host and port as arguments in the `start` function to not load the config here.

**Comment by Geolykt** [comment]
The code looks good to me now, I'll test it later in real use and see whether it works

**Review by Geolykt — DISMISSED** [review]
Crashes:
```
[Geolykt@emeric-pc-linux McPy]$ python main.py 
Traceback (most recent call last):
  File "/home/Geolykt/git/McPy/main.py", line 14, in <module>
    import classes
  File "/home/Geolykt/git/McPy/classes/__init__.py", line 2, in <module>
    from . import Server
  File "/home/Geolykt/git/McPy/classes/Server.py", line 5, in <module>
    from .mcPy.MultiProcessing import MultiProcessing
  File "/home/Geolykt/git/McPy/classes/mcPy/__init__.py", line 1, in <module>
    from .McPy import main
  File "/home/Geolykt/git/McPy/classes/mcPy/McPy.py", line 8, in <module>
    from classes.utils.Config import ConfigParser
  File "/home/Geolykt/git/McPy/classes/utils/__init__.py", line 7, in <module>
    from . import Config
  File "/home/Geolykt/git/McPy/classes/utils/Config.py", line 5, in <module>
    import yaml
ModuleNotFoundError: No module named 'yaml'
[Geolykt@emeric-pc-linux McPy]$
```

**Comment by hydrostaticcog** [comment]
Ah, need to put `yaml` module in requirements.txt. Ig 0/0 did not add it when he added the configparser.

**Review by Geolykt — COMMENTED** [review]
Works fine now, though you can specify negative max playercounts without any warnings; there are probably a few other edge case scenarios out there which I still have to test

**Review by Geolykt — APPROVED** [review]
Does not take malformatted YAML well, otherwise it is remarkably resistant to dangerous input
(testing with `25`, `\"-25` and random text)

**Review by ntoskrnl4 — APPROVED** [review]
If part of the code wasn't already existing then I'd recommend using the vanilla `server.properties` file and format, but otherwise this looks fine to me I guess?

**Comment by hydrostaticcog** [comment]
Named `server_properties.yaml` after @Geolykt 's review.
